File: multi-class-NN/loss_function.js
class loss{
    constructor()
    {
        this.ME;
        this.MSE;
        this.cross_entropy;
    }

    ME = (pre,target) =>
    {
        return pre - target
    }

    MSE = (pre,target) =>
    {
        return Math.pow(pre-target,2)
    }

    cross_entropy = (pre,target) =>
    {
        return -(target * Math.log10(pre) + ((1 - target) * Math.log10(1 - pre)))
    }
}

module.exports = loss

File: loss_function.cpp
#include <cmath>

class Loss {
public:
    Loss() {} // Constructor

    double ME(double pre, double target) {
        return pre - target;
    }

    double MSE(double pre, double target) {
        return std::pow(pre - target, 2);
    }

    double cross_entropy(double pre, double target) {
        return -(target * std::log10(pre) + (1 - target) * std::log10(1 - pre));
    }
};
